daemon/src/version.rs

```diff
@@ -0,0 +1,3 @@
+pub fn version() -> &'static str {
+    env!("VERGEN_GIT_SEMVER_LIGHTWEIGHT")
```

Comment:
Why is this within the `daemon` crate? If we want to centralise it, I think it should go into `shared-bin`.

Reply:
we need to have the access to the `version()` inside the daemon in order to send it across (the additional usage in is the 3rd commit, if we didn't plan to send it, then of course `shared-bin` would be the natural target).

Reply:
Right. Would it make sense to pass it in as a parameter instead? I guess that would add a lot of friction.

Reply:
Wouldn't it be better to use the version from `taker` and `maker` crate (can pass it into the wire actors if necessary)?
I don't think it's important that the versions match, we just want to know what version a taker is running on the maker side.

Reply:
This is the version that is printed on the other side when the logs say "running version XXX" - IMHO it would cause a lot of confusion if we suddenly decided to print a different kind of version. We can later adjust that if there are strong opinions - what matters in this PR most is that we have a string that gets printed :)

Note that the versions from `maker` and `taker` crate actually match this one - with the small addition that if we're running from master, we append SHA hash to the version too.

Reply:
> Right. Would it make sense to pass it in as a parameter instead? I guess that would add a lot of friction.

it would cause more friction and I don't think it's worth it :( can do it later if we decide to.

Reply:
> Note that the versions from `maker` and `taker` crate actually match this one - with the small addition that if we're running from master, we append SHA hash to the version too.

They don't match. The maker and taker crate versions are currently set to `0.1.0`.
As long as we are cutting _repository-wide_ releases (as in we are releasing all binaries based on a single tag), it does not make any sense to version these binaries differently.

There is a proposed ADR for changing the versioning scheme to move away from semantic versioning because exactly what we are doing at the moment is super confusing because it may mean that we are releasing a version of the `taker` binary that has 0 functional changes but because we are cutting repository wide releases, we have to put a release out for `taker` as well even if only `maker` changes.

Reply:
Please, let's not dive into versioning details in this issue here. To me hhe comment above just supports that I'm printing the correct one and highlights troubles with having *multiple* version numbers above.

Reply:
@da-kami would you prefer me to print out `0.1.0` here?

Reply:
> @thomaseizinger , let's not dive into versioning debate in this issue here. To me the comment above just supports that I'm printing the correct one (the one that matters here). I am yet to see any usefulness from having multiple versions - but the problems highlighted by me in the ADR are clearly visible already :(

I totally agree! I don't see us moving away from a single repository any time soon and until then, repository-wide releases make total sense which also means, everything should have one version number.

What I was pointing out is that printing the _crate_ version of the maker and taker binary is pointless because it is set to `0.1.0` and that is _different_ from the version we tag.